### PR TITLE
Add proper comparison for sort based on data types

### PR DESCRIPTION
**Pull request by vasukalariya**
# Description
Summarize the change.
- Added proper sorting based on datatypes.
- Backfilled values should always be present at last.

Fixes #<issue-number> (link all the GitHub issues this addresses)

# Testing
Describe how you tested this code. How can the reviewers reproduce your tests?
- Added unit test for sorting.
- Validated the same using the query `app_name=Bracecould | sort variable_col_4 | fields variable_col_4`. The filter has 1 numeric value, 3 string values in form of numeric and 2 backfilled values.
- Also verified tiebreaker sort using queries like `app_name=Bracecould | fields http_status, country | sort http_status, -country `.

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [ ] I have self-reviewed this PR.
- [ ] I have removed all print-debugging and commented-out code that should not be merged.
- [ ] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [ ] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: pkg/segment/query/processor/sortcommand_test.go [hunk 1]
```diff
@@ -278,3 +278,107 @@ func Test_SortCommand_withRRCs(t *testing.T) {
...
+		},
+	}
+
+	expected := []utils.CValueEnclosure{
+		{Dtype: utils.SS_DT_STRING, CVal: "1"},
+		{Dtype: utils.SS_DT_STRING, CVal: "2"},
+		{Dtype: utils.SS_DT_FLOAT, CVal: float64(3)},
+		{Dtype: utils.SS_DT_SIGNED_NUM, CVal: int64(4)},
+		{Dtype: utils.SS_DT_UNSIGNED_NUM, CVal: uint64(5)},
+		{Dtype: utils.SS_DT_STRING, CVal: "A"},
```
Comment (AndrewHess) [inline pkg/segment/query/processor/sortcommand_test.go:322]: nit: add a two-digit string number to this test (e.g., would `"15"` be between `"1"` and `"2"` or after `5`?)

Are we handling all these mixed types (string, int, bool), the same way Splunk does?

File: pkg/segment/query/processor/sortcommand.go [hunk 2]
```diff
@@ -96,6 +99,131 @@ func (p *sortProcessor) Process(inputIQR *iqr.IQR) (*iqr.IQR, error) {
...
+	OTHER
+)
+
+type COMPARE uint8
+
+const (
+	EQUAL = iota
+	LESS
+	GREATER
+)
```
[the remark this answers is not in the document]
Comment (vasukalariya) [reply pkg/segment/query/processor/sortcommand.go]: addressed in the latest commit.

File: pkg/segment/query/processor/sortcommand.go [hunk 3]
```diff
@@ -96,6 +99,131 @@ func (p *sortProcessor) Process(inputIQR *iqr.IQR) (*iqr.IQR, error) {
...
+			floatValB, isFloat := valueB.GetFloatValueIfPossible()
+			if !isFloat {
+				return LESS
+			}
+			result = compareFloat(floatValA, floatValB)
+		case STRING:
+			strValA, err := valueA.GetStringForGroupByCol()
+			if err != nil {
+				return GREATER
+			}
```
[the remark this answers is not in the document]
Comment (vasukalariya) [reply pkg/segment/query/processor/sortcommand.go]: addressed in the latest commit.

File: pkg/segment/query/processor/sortcommand.go [hunk 4]
```diff
@@ -96,6 +99,131 @@ func (p *sortProcessor) Process(inputIQR *iqr.IQR) (*iqr.IQR, error) {
...
+	}
+}
+
+// Returns A comparison B
+func compare(valueA, valueB *segutils.CValueEnclosure, asc bool) COMPARE {
+	var result COMPARE
+	rankA := getRank(valueA)
+	rankB := getRank(valueB)
+
+	// OTHER rank records always be at the end
```
[the remark this answers is not in the document]
Comment (vasukalariya) [reply pkg/segment/query/processor/sortcommand.go]: addressed in the latest commit.

File: pkg/segment/query/processor/sortcommand.go [hunk 5]
```diff
@@ -96,6 +99,131 @@ func (p *sortProcessor) Process(inputIQR *iqr.IQR) (*iqr.IQR, error) {
...
+		return LESS
+	}
+
+	if rankA != rankB {
+		if rankA < rankB {
+			result = COMPARE(LESS)
+		} else {
+			result = COMPARE(GREATER)
+		}
+	} else {
```
[the remark this answers is not in the document]
Comment (vasukalariya) [reply pkg/segment/query/processor/sortcommand.go]: addressed in the latest commit.

File: pkg/segment/query/processor/sortcommand_test.go (hunk 1 above)
Comment (vasukalariya) [reply pkg/segment/query/processor/sortcommand_test.go]: addressed in the latest commit. Yes, its similar to what they do.

**Review by AndrewHess — APPROVED** [review]

**Pull request merged**